pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -281,16 +288,17 @@ protected void internalDeleteNamespace(AsyncResponse asyncResponse, boolean auth
             if (log.isDebugEnabled()) {
                 log.debug("Found topics on namespace {}", namespaceName);
             }
-            boolean hasNonSystemTopic = false;
-            for (String topic : topics) {
+            for (String topic : partitionedTopicNames) {
                 if (!pulsar().getBrokerService().isSystemTopic(TopicName.get(topic))) {
-                    hasNonSystemTopic = true;
-                    break;
+                    asyncResponse.resume(new RestException(Status.CONFLICT, "Cannot delete non empty namespace"));
```

Comment:
What about logging a line that reports the topic that is preventing the deletion, here and below?

Reply:
Good idea. Already add log line